### PR TITLE
Ensure embassy-lora stm32wl supports log crate

### PR DESCRIPTION
**Pull request by lulf**

### CODE REVIEW

File: embassy-lora/src/stm32wl/mod.rs [hunk 1]
```diff
@@ -172,7 +172,7 @@ impl<'d, RS: RadioSwitch> SubGhzRadio<'d, RS> {
                 self.radio.read_buffer(ptr, &mut buf[..len as usize])?;
                 self.radio.set_standby(StandbyClk::Rc)?;
 
-                trace!("RX done: {=[u8]:#02X}", &mut buf[..len as usize]);
+                trace!("RX done: {:02x?}", &mut buf[..len as usize]);
```
Comment (Dirbaio) [inline embassy-lora/src/stm32wl/mod.rs:175]: defmt needs `:02x`, log needs `:02x?`

Very unfortunate https://github.com/knurling-rs/defmt/issues/648

Comment (lulf) [reply embassy-lora/src/stm32wl/mod.rs:175]: 😭

Comment (lulf) [reply embassy-lora/src/stm32wl/mod.rs:175]: what's the general preferred workaround? #[cfg()] or just derive LowerHex (if that works?)

Comment (Dirbaio) [reply embassy-lora/src/stm32wl/mod.rs:175]: I've done cfg's for this in the past... A wrapper type deriving defmt and LowerHex could work but I think it's more annoying...

**Pull request merged**